redash/models.py

```diff
@@ -1000,6 +1007,29 @@ def outdated_queries(cls):
 
         return outdated_queries.values()
 
+    @classmethod
+    def delete_stale_resultsets(cls):
+        delete_count = 0
+        queries = Query.query.filter(Query.schedule_keep_results != None).order_by(Query.schedule_keep_results.desc())
+        if queries.first() and queries[0].schedule_keep_results:
```

Comment:
Maybe assign the result of `first()` to a variable and use it in the condition:
`if query is not None and query.schedule_resultset_size > 0`: